### PR TITLE
refactor(colorpickers, grid, loaders, notifications, tags tooltips): use transient props where appropriate 

### PR DESCRIPTION
**Pull request by ze-flo**
## Description
This PR updates various components in `react-colorpickers`, `react-grid`, `react-notifications`, `react-loaders`, and `react-tooltips` to use [transient props](https://styled-components.com/docs/api#transient-props) where appropriate. These changes are necessary in  preparation for the upgrade to `styled-components@6.x.x` to ensure we avoid DOM violation errors after the transition.

## Checklist

- [ ] ~~:ok_hand: design updates will be Garden Designer approved (add the designer as a reviewer)~~
- [x] :globe_with_meridians: demo is up-to-date (`npm start`)
- [ ] ~~:arrow_left: renders as expected with reversed (RTL) direction~~
- [ ] ~~:black_circle: renders as expected in dark mode~~
- [ ] ~~:metal: renders as expected with Bedrock CSS (`?bedrock`)~~
- [x] :guardsman: includes new unit tests. Maintain existing coverage (always >= 96%)
- [ ] ~~:wheelchair: tested for WCAG 2.1 AA accessibility compliance~~
- [ ] ~~:memo: tested in Chrome, Firefox, Safari, and Edge~~

### CODE REVIEW

File: packages/loaders/src/elements/Dots.tsx [hunk 1]
```diff
@@ -61,18 +61,18 @@ export const Dots = forwardRef<SVGSVGElement, IDotsProps>(
 
     return (
       <StyledSVG
+        $dataGardenId={COMPONENT_ID}
         ref={ref}
         fontSize={size!}
         color={color!}
         width="80"
         height="72"
```
Comment (jzempel) [inline packages/loaders/src/elements/Dots.tsx:69]: I don't think we want these props leaking to the DOM. Also, might need to `git-blame`, but why do we have these huge `width`/`height` values?

File: packages/loaders/src/elements/Inline.tsx [hunk 2]
```diff
@@ -28,7 +28,7 @@ export const Inline = forwardRef<SVGSVGElement, IInlineProps>(({ size, color, ..
     // eslint-disable-next-line jsx-a11y/prefer-tag-over-role
     <StyledInline
       ref={ref}
-      size={size!}
+      $size={size!}
       color={color!}
```
Comment (jzempel) [inline packages/loaders/src/elements/Inline.tsx:32]:
```suggestion
      $color={color!}
```

File: packages/loaders/src/styled/StyledSVG.ts [hunk 3]
```diff
@@ -9,13 +9,13 @@ import styled, { css, DefaultTheme, ThemeProps } from 'styled-components';
 import { getColor, retrieveComponentStyles } from '@zendeskgarden/react-theming';
 
 interface IStyledSVGProps {
-  dataGardenId: string;
+  $dataGardenId: string;
   color?: string;
   fontSize?: string | number;
   width: number | string;
   height: number | string;
```
Comment (jzempel) [inline packages/loaders/src/styled/StyledSVG.ts:16]:
```suggestion
  $color?: string;
  $fontSize?: string | number;
  $width: number | string;
  $height: number | string;
```

We can remove these from the DOM since they only need to inform the CSS

File: packages/loaders/src/elements/Dots.tsx [hunk 4]
```diff
@@ -61,18 +61,18 @@ export const Dots = forwardRef<SVGSVGElement, IDotsProps>(
 
     return (
       <StyledSVG
+        $dataGardenId={COMPONENT_ID}
```
Comment (jzempel) [inline packages/loaders/src/elements/Dots.tsx:64]:
```suggestion
        dataGardenId={COMPONENT_ID}
```

The `data-garden-id` needs to remain intact in the DOM

File: packages/loaders/src/styled/StyledSVG.ts [hunk 5]
```diff
@@ -9,13 +9,13 @@ import styled, { css, DefaultTheme, ThemeProps } from 'styled-components';
 import { getColor, retrieveComponentStyles } from '@zendeskgarden/react-theming';
 
 interface IStyledSVGProps {
-  dataGardenId: string;
+  $dataGardenId: string;
```
Comment (jzempel) [inline packages/loaders/src/styled/StyledSVG.ts:12]:
```suggestion
  dataGardenId: string;
```

**Review by jzempel — CHANGES_REQUESTED** [review]

File: packages/loaders/src/elements/Dots.tsx (hunk 4 above)
Comment (ze-flo) [reply packages/loaders/src/elements/Dots.tsx]: `dataGardenId` would be filtered out by [@emotion/is-prop-valid](https://www.npmjs.com/package/@emotion/is-prop-valid) used under the hood in `styled-components@v5`. I've replaced it with `data-garden-id`.

Screenshot 2024-11-06 at 8 03 16 AM

File: packages/dropdowns.legacy/src/styled/multiselect/StyledMultiselectInput.ts [hunk 6]
```diff
@@ -13,22 +13,26 @@ const COMPONENT_ID = 'dropdowns.multiselect_input';
 
 interface IStyledMultiselectInputProps {
   isCompact: boolean;
```
Comment (jzempel) [inline packages/dropdowns.legacy/src/styled/multiselect/StyledMultiselectInput.ts:15]: Should this be `$isCompact`?

File: packages/loaders/src/elements/Dots.tsx [hunk 7]
```diff
@@ -61,18 +61,18 @@ export const Dots = forwardRef<SVGSVGElement, IDotsProps>(
...
     return (
       <StyledSVG
+        data-garden-id={COMPONENT_ID}
         ref={ref}
-        fontSize={size!}
-        color={color!}
+        $fontSize={size!}
+        $color={color!}
         width="80"
         height="72"
```
Comment (jzempel) [inline packages/loaders/src/elements/Dots.tsx:69]: I still think something about this has a bad smell. As a rule, we don't put `width`/`height` attributes on SVGs. Rather, the preference is to apply sizing via CSS. Not sure how/why these got here.

File: packages/loaders/src/elements/Spinner.tsx [hunk 8]
```diff
@@ -74,19 +74,19 @@ export const Spinner = forwardRef<SVGSVGElement, ISpinnerProps>(
 
     return (
       <StyledSVG
+        $color={color}
+        $containerHeight="1em"
+        $containerWidth="1em"
+        $fontSize={size}
+        data-garden-id={COMPONENT_ID}
+        height={HEIGHT}
```
Comment (jzempel) [inline packages/loaders/src/elements/Spinner.tsx:82]: ditto re: `height` and `width`. These should be CSS-only

File: packages/loaders/src/styled/StyledSVG.ts [hunk 9]
```diff
@@ -9,17 +9,17 @@ import styled, { css, DefaultTheme, ThemeProps } from 'styled-components';
...
 
 interface IStyledSVGProps {
-  dataGardenId: string;
-  color?: string;
-  fontSize?: string | number;
+  'data-garden-id': string;
+  $color?: string;
+  $fontSize?: string | number;
   width: number | string;
   height: number | string;
```
Comment (jzempel) [inline packages/loaders/src/styled/StyledSVG.ts:16]: ditto re: `width` and `height`

File: packages/pagination/src/styled/OffsetPagination/StyledPageBase.ts [hunk 10]
```diff
@@ -69,6 +69,12 @@ const colorStyles = ({ theme }: ThemeProps<DefaultTheme>) => {
       inset: true
     })}
 
+    &:disabled,
+    [aria-disabled='true'] {
+      background-color: transparent;
+      color: ${disabledColor};
+    }
```
Comment (jzempel) [inline packages/pagination/src/styled/OffsetPagination/StyledPageBase.ts:76]: Why was this moved? The result is that `:active` styling takes precedent over `:disabled`, which is not correct.

File: packages/pagination/src/styled/OffsetPagination/StyledPageBase.ts [hunk 11]
```diff
@@ -69,6 +69,12 @@ const colorStyles = ({ theme }: ThemeProps<DefaultTheme>) => {
       inset: true
     })}
 
+    &:disabled,
+    [aria-disabled='true'] {
```
Comment (jzempel) [inline packages/pagination/src/styled/OffsetPagination/StyledPageBase.ts:73]:
```suggestion
    &[aria-disabled='true'] {
```

File: packages/tags/demo/stories/TagStory.spec.tsx
Comment (jzempel) [inline packages/tags/demo/stories/TagStory.spec.tsx:1]: I thought we weren't committing to snapshot testing. I don't think it has long-term value here.

File: packages/tags/demo/stories/__snapshots__/TagStory.spec.tsx.snap
Comment (jzempel) [inline packages/tags/demo/stories/__snapshots__/TagStory.spec.tsx.snap:1]: Recommend delete

File: packages/loaders/src/elements/Dots.tsx (hunk 4 above)
Comment (jzempel) [reply packages/loaders/src/elements/Dots.tsx]: Hmm, bit of a bummer about that filtering.

File: packages/dropdowns.legacy/src/styled/multiselect/StyledMultiselectInput.ts (hunk 6 above)
Comment (ze-flo) [reply packages/dropdowns.legacy/src/styled/multiselect/StyledMultiselectInput.ts:15]: No in this case. This case is tricky to understand. Using `$isCompact` led to a snapshot mismatch.

TL&DR;

`StyledMultiselectInput` extends `StyledInput`, which in turns extends `Input`, a `React FC` that returns a `StyledTextInput`. `Input` only accepts `isCompact`, which is received by `StyledTextInput` after being renamed as transient prop `$isCompact`.

**Review by jzempel — APPROVED** [review]
